packages/core/strapi/lib/core/registries/custom-fields.js

```diff
@@ -56,6 +56,23 @@ const customFieldsRegistry = (strapi) => {
           throw new Error(`Custom field name: '${name}' is not a valid object key`);
         }
 
+        // Validate inputSize when provided
+        if (inputSize) {
+          if (
+            !isObject(inputSize) ||
```

Comment:
you're right. The error message is just a bit nicer is we can catch non-object values explicity, instead of the one you'd get when reading the properties. I added an additional `Array.isArray` check to be safe

Reply:
Nice! I'll do a functional test for QA this afternoon before approving